src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,281 @@
+            sum = ((op1 % op2) - op2)
+        else:
+            sum = (op1 % op2)
+        return {'oper': '%', 'op2': op2, 'sum': sum}
+
+    def bitshiftleft(self, op1, op2):
+        op2 = int(math.fabs(op2))
+        if op1 < 0:
+            op2 = math.floor(math.log(MIN/op1))
+        else:
+            op2 = math.ceil(math.log(MIN/op1))
+        sum = op1 << op2
+        if sum < MIN:
+            if op1 < 0:
+                op2 = math.floor(math.log(MIN/op1, 2))
+            else:
+                op2 = math.ceil(math.log(MIN/op1, 2))
+        elif sum > MAX:
+            if op1 > 0:
+                op2 = math.floor(math.log(MAX/op1, 2))
+            else:
+                op2 = math.ceil(math.log(MAX/op1, 2))
+        sum = op1 << op2
+        return {'oper': '<<', 'op2': op2, 'sum': sum}
+
+    def bitand(self, op1, op2):
+        sum = op1 & op2
+        if sum < MIN:
+            op2 = sum | op1
+        elif sum > MAX:
+            print("bigger")
+        sum = op1 & op2
+        return {'oper': '&', 'op2': op2, 'sum': sum}
+
+    def bitxor(self, op1, op2):
+        if op1 > 0:
+            op2 = int(math.fabs(op2))
+        else:
+            op2 = - int(math.fabs(op2))
+        sum = op2 ^ op1
+        return {'oper': '^', 'op2': op2, 'sum': sum}
+
+    def debug(self):
+        last = self.operands[0]
+        for i in range(self.num_operands -1):
+            result = self.operators_list[self.operators[i]](last, self.operands[i+1])
+            print("%d. %d %s %d == %d" % (i, last, result["oper"], result["op2"], result["sum"]))
+            self.operands[i+1] = result["op2"]
+            last = result["sum"]
+
+    def printJS(self):
+        last = self.operands[0]
+        print("%s%d" % ("(" * (self.num_operands -1), last), end='')
+        for i in range(self.num_operands -1):
+            result = self.operators_list[self.operators[i]](last, self.operands[i+1])
+            print(" %s %d)" % (result["oper"], result["op2"]), end='')
+            self.operands[i+1] = result["op2"]
+            last = result["sum"]
+        print(" == %d" % last,)
+
+
+
+# gs = GenSequence(15)
+# gs.debug()
+# gs.printJS()
+def generate_binary_tests(options):
+    if not os.path.isdir(options.output):
+        os.makedirs(options.output)
+    SEED = options.seed[0]
+    random.seed(SEED)
+    counter = 0
+    file_count = 1
+    new_file_begin = True
+    bitcalc = False
+    arithcalc = False
+    for i in range(len(options.op)):
+        if ((options.op[i] == "+") or (options.op[i] == "-") or (options.op[i] == "*") or (options.op[i] == "/") or
+        (options.op[i] == "%")):
+            arithcalc = True
+        elif (options.op[i] == "^"):
+            bitcalc = True
+
+    file_to_open = os.path.join(options.output, filename.format(NUMBER = file_count))
+    for i in range(options.test_count[0]):
+        bit = False
+        counter = counter + 1
+        if (counter % 800 == 0):
```

Comment:
Where does this `800` come from? Please do not use magic numbers.

Reply:
Fixed